kolibri/core/assets/src/heartbeat.js

```diff
@@ -0,0 +1,58 @@
+const logging = require('kolibri.lib.logging').getLogger(__filename);
+
+class HeartBeat {
+  constructor(kolibri) {
+    this.kolibri = kolibri;
+    // Do this to have a consistent callback that has 'this' properly bound
+    // but can be repeatedly referenced to add and remove listeners.
+    this.setActiveCallback = this.setActive.bind(this);
+    this.setInactive();
+    this.start();
+  }
+  start() {
+    logging.debug('Starting heartbeat');
+    this.setActivityListeners();
+    this.beat();
+  }
+  setActivityListeners() {
+    this.events.forEach((event) => {
+      document.addEventListener(event, this.setActiveCallback, true);
+    });
+  }
+  clearActivityListeners() {
+    this.events.forEach((event) => {
+      document.removeEventListener(event, this.setActiveCallback, true);
+    });
+  }
+  setActive() {
+    this.active = true;
+    this.clearActivityListeners();
+  }
+  setInactive() {
+    this.active = false;
+  }
+  beat() {
+    if (this.active) {
```

Comment:
If `this.active` was toggled to `true` at any point in this interval, there was some user activity, so we can let the server know by polling the SessionModel.